app/models/concerns/thredded/topic_common.rb

```diff
@@ -47,24 +47,45 @@ def unread(user)
           .merge(reads_class.where(reads[:id].eq(nil).or(reads[:read_at].lt(topics[:updated_at]))))
       end
 
+      private
+
+      # @param user [Thredded.user_class]
+      # @return [ByPostableLookup]
+      def read_states_by_topics_lookup(user, null_value)
+        read_states = reflect_on_association(:user_read_states).klass
+          .where(user_id: user.id, postable_id: current_scope.map(&:id))
+        Thredded::TopicCommon::Lookup.new(read_states, null_value)
```

Comment:
@timdiggins I think the hash solution can be made clearer with better variable/method names and maybe a comment. I prefer it to the class one because a simple pre-populated data structure is easier to use than a custom class, because Hash methods are well-known. Either way is fine by me though.